python/tests/sklearn/preprocessing/test_one_hot_encoder.py

```diff
@@ -0,0 +1,145 @@
+        ohe = OneHotEncoder(handle_unknown='error')
+        ohe.mlinit(prior_tf=self.le, output_features='{}_one_hot_encoded'.format(self.le.output_features))
+        ohe.fit(self.oh_data)
+
+        ohe.serialize_to_bundle(self.tmp_dir, ohe.name)
+        with open("{}/{}.node/model.json".format(self.tmp_dir, ohe.name)) as json_data:
+            model = json.load(json_data)
+
+        self.assertEqual('one_hot_encoder', model['op'])
+        self.assertEqual(3, model['attributes']['size']['long'])
+        self.assertEqual('error', model['attributes']['handle_invalid']['string'])
+        self.assertEqual(False, model['attributes']['drop_last']['boolean'])
+
+    def test_one_hot_encoder_deserialization_succeeds_when_handle_unknown_is_set_to_error(self):
+        ohe = OneHotEncoder(handle_unknown='error')
+        ohe.mlinit(prior_tf=self.le, output_features='{}_one_hot_encoded'.format(self.le.output_features))
+        ohe.fit(self.oh_data)
+
+        ohe.serialize_to_bundle(self.tmp_dir, ohe.name)
+
+        node_name = "{}.node".format(ohe.name)
+        ohe_ds = OneHotEncoder()
+        ohe_ds.deserialize_from_bundle(self.tmp_dir, node_name)
+
+        self.oh_data[2][0] = 3  # Add an unknown category
+
+        with self.assertRaises(ValueError):
+            ohe_ds.transform(self.oh_data)
+
+    def test_one_hot_encoder_serialization_succeeds_when_handle_unknown_is_set_to_ignore(self):
+        labels = ['a', 'b', 'c', 'a', 'b', 'b']
+
+        le = LabelEncoder(input_features=['label'], output_features='label_le_encoded')
+        oh_data = le.fit_transform(labels).reshape(-1, 1)
+
+        ohe = OneHotEncoder(handle_unknown='ignore')
+        ohe.mlinit(prior_tf=le, output_features='{}_one_hot_encoded'.format(le.output_features))
+        ohe.fit(oh_data)
+
+        ohe.serialize_to_bundle(self.tmp_dir, ohe.name)
+        with open("{}/{}.node/model.json".format(self.tmp_dir, ohe.name)) as json_data:
+            model = json.load(json_data)
+
+        self.assertEqual('one_hot_encoder', model['op'])
+        self.assertEqual(3, model['attributes']['size']['long'])
+        self.assertEqual('keep', model['attributes']['handle_invalid']['string'])
+        self.assertEqual(True, model['attributes']['drop_last']['boolean'])
+
+    def test_one_hot_encoder_deserialization_succeeds_when_handle_unknown_is_set_to_ignore(self):
+        ohe = OneHotEncoder(handle_unknown='ignore')
+        ohe.mlinit(prior_tf=self.le, output_features='{}_one_hot_encoded'.format(self.le.output_features))
+        ohe.fit(self.oh_data)
+
+        ohe.serialize_to_bundle(self.tmp_dir, ohe.name)
+
+        node_name = "{}.node".format(ohe.name)
+        ohe_ds = OneHotEncoder()
+        ohe_ds.deserialize_from_bundle(self.tmp_dir, node_name)
+
+        self.oh_data[2][0] = 3  # Add an unknown category
+
+        expected = ohe.transform(self.oh_data).todense()
+        actual = ohe_ds.transform(self.oh_data)
```

Comment:
Why do we need to call toDense() on ohe.transform but not on ohe_ds.transform, I would expect the model behavior to be the same pre and post serialization.

Reply:
Good question! The MLeap deserialization code for OneHotEncoder has always set `sparse=False` when loading a bundle. 

https://github.com/combust/mleap/blob/d703c8dd5e7b93c7f6c8b45fa6c9aadb9c026a92/python/mleap/sklearn/preprocessing/data.py#L658

I believe it's to make it easy to convert into a dataframe in this function:

https://github.com/combust/mleap/blob/d703c8dd5e7b93c7f6c8b45fa6c9aadb9c026a92/python/mleap/sklearn/preprocessing/data.py#L778

Because otherwise you would have to call `from_spmatrix`:

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.sparse.from_spmatrix.html

But I'm not 100% sure this is the reason why

Either way, the default for the scikit-learn OneHotEncoder is to set `sparse=True`. So it seems that this normally results in an implicit conversion. Another option would be to error unless users explicitly set `sparse=False` every time they instantiate a OneHotEncoder, but that would be pretty annoying. I'm not sure if you have an opinion on which is better...